core/tests/performance_framework/perf_services.py

```diff
@@ -56,15 +58,48 @@ class SeleniumPerformanceDataFetcher(object):
 
     # Duration (in seconds) to wait for the complete page to load, otherwise
     # XHR requests made post initial page load will not be recorded.
-    WAIT_DURATION_SECS = 3
-
-    def __init__(self, browser=DEFAULT_BROWSER_SOURCE):
+    DEFAULT_WAIT_DURATION_SECS = 3
+
+    BASE_URL = 'http://localhost:%d' % feconf.PERFORMANCE_TESTS_SERVER_PORT
+
+    def __init__(self, browser=DEFAULT_BROWSER_SOURCE, username=None,
+                 do_login=False, create_exploration=False,
+                 reload_demo_collections=False, reload_demo_explorations=False,
+                 reload_first_exploration=False):
+        """preload_options:
+            login: True if login is required.
+            enable_explorations: True if explorations needs to be enabled.
+            enable_collections: True if collections needs to be enabled.
+            create_exploration: True if exploration needs to be created.
+        """
         if browser in self.SUPPORTED_BROWSER_SOURCES:
             self.browser = browser
         else:
             error_msg = 'Unsupported browser specified: %s' % browser
             raise ValueError(error_msg)
 
+        self.exploration_url = None
+        self.dev_appserver_login_cookie = None
+
+        # All tests are independent and they do login + signup + other test
+        # specific work. Using the same username across tests breaks this flow.
+        # Thus, different usernames enable independent tests.
+        self.email = '%s@example.com' % username
+        self.username = username
+
+        driver = self._setup_driver()
+        if do_login:
+            self._setup_login(driver)
+        elif create_exploration:
+            self._create_exploration(driver)
+        elif reload_demo_collections:
+            self._setup_reload_demo_collections(driver)
+        elif reload_demo_explorations:
+            self._setup_reload_demo_explorations(driver)
+        elif reload_first_exploration:
+            self._setup_reload_first_exploration(driver)
+        self._stop_driver(driver)
```

Comment:
It's worth adding a comment here to say why we do this (as you explained to me in the response to the other comment I left here).